test/integration/smoke/test_deploy_vm_extra_config_data.py

```diff
@@ -294,3 +300,67 @@ def test_deploy_vm__with_extraconfig_vmware(self):
             )
 
         self.destroy_vm(response.id)
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_deploy_vm__with_extraconfig_xenserver(self):
+        if self.hypervisor.lower() != 'xenserver':
+            raise self.skipTest("Skipping test case for non-xenserver hypervisor")
+        """
+        Following commands are used to convert a VM from HVM to PV
+        "vm-param-set HVM-boot-policy=""
+         vm-param-set PV-bootloader=pygrub
+         vm-param-set PV-args="hvc0"
+        """
+        extraconfig = "%22vm-param-set%20HVM-boot-policy%3D%22%22%0Avm-param-set%20PV-bootloader%3Dpygrub%0Avm-param-set%20PV-args%3D%22hvc0%22"
+        response = self.deploy_vm(extraconfig)
+        host_id = response.hostid
+        host = list_hosts(
+            self.apiclient,
+            id=host_id)
+
+        host_ipaddress = host[0].ipaddress
+
+        ssh_client = SshClient(host_ipaddress, port=22,
+                               user=self.hostConfig['username'],
+                               passwd=self.hostConfig['password'])
+
+        extraconfig_decoded = urllib.unquote(extraconfig)
+        config_arr = extraconfig_decoded.splitlines()
+
+        # Get vm instance uuid
+        instance_uuid = self.get_vm_uuid(response.instancename, ssh_client)
+        try:
+            for config in config_arr:
+                config_tuple = self.get_xen_param_values(config)
+                vm_config_check = 'xe vm-param-get param-name={} uuid={}'.format(config_tuple[0], instance_uuid)
+                result = ssh_client.execute(vm_config_check)
+                param_value = config_tuple[1].strip()
+                if param_value == '""':
+                    result[0] = '""';
```

Comment:
Maybe the semicolon here is not needed

Reply:
True, I will remove them. I was probably mixing Java with Python.